### DISCUSSION TITLE
[Request] Setting to default "Hide Whitespace" to true unless diff contains only whitespace changes

### DISCUSSION THREAD

**Discussion by RobvH** [community/community#26629] [Pull Requests]
Remove a while() loop from around a block of code and the subsequent indentation changes that causes wreaks havoc on diff's ability to accurately portray the code changes. Generally, the while loop change will be mostly lost and tons of non-existent changes will be shown because diff is confused. Fantastically, there's a feature for that!!

Not so fantastically, you have to know (and few do) that "hide whitespace" is a thing. Further, you can't make it the default. 

The request: perhaps a user "Appearance" setting that defaults diff views to "hide whitespace" being on. If you want to get nuanced, you could flip that if there's only whitespace changes.

Seriously, if you're having whitespace discussions in your PRs you need a linter and lint check in your workflow. Mostly, showing whitespace changes to diff is just causing confusion and burning extra brain (and cpu) cycles. 

Thanks for taking the time to take this word-journey (wordney?) with me. I hope this suggestion makes you go "yeah! OMG why weren't we already doing this? cause we ALL know the pain the causes!" If not, I hope you'll still consider it. 😅

**Comment by alejunco** [comment]
If you are using Chrome give it a try to https://chrome.google.com/webstore/detail/github-whitespace/fnpkdafamnbjoldglihkjjdicofghccm

**Comment by VersenyiZsolt** [comment]
Im working on fixing this type of problems with the diff viewer. Try it out to, really curious about your opinion.

[www.archdiff.com](www.archdiff.com)
